### DISCUSSION TITLE
.MTS videos won't load (infinite loading)

### DISCUSSION THREAD

**Discussion by Pheggas** [photoprism/photoprism#2318] [General]
Hello. I just started using Photoprism. Indexed all photos and videos as well. I decided to import videos from standalone camera i had which were encoded in .MTS format. I found on [Photoprism site that MTS is supported format](https://photoprism.app/kb/file-formats) so i don't really understand why it won't play. In Library -> Logs there's nothing to show when i try to load it.

In NAS those videos are perfectly accessible as well as on Photoprism's host VM. Is there any additional log so i can find more info about this issue?

PS, export of video info from MediaTab:
===================== General =====================
Id                          : 1 (0x1)
Complete name               : path
Format                      : BDAV
Format info                 : Blu-ray Video
File size                   : 693 MB
Duration (ms)               : 5mn 58s
Overall bit rate mode       : Variable
Overall bit rate            : 16.2 Mbps
Maximum Overall bit rate    : 18.0 Mbps

===================== Video =====================
Id                          : 4113 (0x1011)
Menu ID                     : 1 (0x1)
Format                      : AVC
Format info                 : Advanced Video Codec
Formatprofile               : High@L4
Formatsettings, CABAC       : Yes
Formatsettings, ReFrames    : 4 frames
Formatsettings, GOP         : M=3, N=12
Codec Id                    : 27
Duration (ms)               : 5mn 58s
Bit rate mode               : Variable
Bit rate                    : 15.3 Mbps
Maximum bit rate            : 16.8 Mbps
Width                       : 1 920 pixels
Height                      : 1 080 pixels
Display aspect ratio        : 16:9
Frame rate                  : 25.000 fps
Color space                 : YUV
Chroma subsampling          : 4:2:0
Bit depth                   : 8 bits
Scan type                   : Interlaced
Scan type, store method     : Separated fields
Scan order                  : Top Field First
Bits/(Pixel*Frame)          : 0.295
Stream size                 : 654 MB (94%)

===================== Audio =====================
Id                          : 4352 (0x1100)
Menu ID                     : 1 (0x1)
Format                      : AC-3
Format info                 : Audio Coding 3
Mode extension              : CM (complete main)
Formatsettings, Endianness  : Big
Codec Id                    : 129
Duration (ms)               : 5mn 58s
Bit rate mode               : Constant
Bit rate                    : 256 Kbps
Channel(s)                  : 2 channels
Channel positions           : Front: L R
Sampling rate               : 48.0 KHz
Frame rate                  : 31.250 fps (1536 spf)
Compression mode            : Lossy
Delayrelative to video      : -80ms
Stream size                 : 10.9 MB (2%)

===================== Text =====================
Id                          : 4608 (0x1200)
Menu ID                     : 1 (0x1)
Format                      : PGS
Codec Id                    : 144
Duration (ms)               : 5mn 58s
Delayrelative to video      : -80ms

**Comment by lastzero** [comment]
You can enable trace mode to see the commands executed for transcoding in the background including any error messages.

Since MTS uses AVC internally, it could be streamed directly to the browser, but isn't really supported there due to the differences in container format (not codec). I haven't tested this much, as MTS is a rather unusual format unless you have a camcorder or rip DVDs/Blu-Ray discs. Make sure to test with multiple browsers, as they may behave differently.

**Reply by Pheggas** [reply]
So i added PHOTOPRISM_TRACE to docker compose file and started the stack (i'm using portainer for managing docker containers / stacks). I played the problematic videos in multiple browsers (with different engines - Chrome, Edge, Firefox, Opera) but none of them actually played the video. I headed into logs (both Library -> Logs and also Portainer -> Photoprism container -> logs) and both of them were showing something with h264_vaapi can't transcode the video. I wanted to passthrough iGPU so i set encoder to h264_vaapi.

Anyway, the weird part comes right now. After checking the logs and seeing that h264_vaapi wasn't able to transcode the video, i headed onto "Search" page which content's wasn't loading at all. From now on, CPU went to 100% workload. I checked logs with trace turned on to see what the hell is going on. There was literally nothing. No new logs, messages, anything. So i turned off whole stack, disabled trace (i assumed it was caused by trace logging as before enabling that option, everything was normal), then i turned on whole stack again and went to Search again. CPU went again absolutely crazy, logs showed nothing. I turned off the stack again, disabled all of the GPU stuff (Photoprism INIT, FFMPEG_BUFFERS, BITRATE, ENCODER and also disabled whole devices section). Turned on the stack and untill i loaded the site, cpu was idling at 0,2-ish %. When i went on the site, cpu went crazy again and then i closed the tab. Were waiting for like a minute and the CPU changed usage from 100% to exactly 50% and were idling on that numbers.

Right now, i'm not sure what i screwed up as the initial crazyness were caused after the PHOTOPRISM_TRACE line was added.

Firstly i need to get Photoprism working normally again and then i can tweak settings to get the playback working.

**Any clues?**

**Reply by Pheggas** [reply]
### **This comment is partly a timeline:**

I enabled the whole GPU stuff again and trace too. Now it's idling at exactly 33% continuously, "Search" content's won't load, after minute or so of loading, the "Offline" warking above the Admin account appeared like this:
![image](https://user-images.githubusercontent.com/9994511/168275172-20f2b8f0-8fbf-46ee-847f-85d6b03154d3.png)

After like 5 seconds it disappears. After another 2-ish minutes it appeared and disappeared again.
Logs in portainer:
![image](https://user-images.githubusercontent.com/9994511/168275490-0f459ba7-ebe9-4c8c-959f-5955ea3431df.png)

(each 1:30 minutes it goes offline in Web UI and online again)
I tried disabling debug logs in Web UIs settings. After then, every setting went gray and can't turn any of it.

After another 3 minutes, the notification "Settings saved" appeared at the bottom middle of the Web UI. I just realised the count of photos at "Search, Albums, Videos, People, Moments, Calendar, Places, Labels, Folders, Private" appeared.

After another minute the notification "More than 50 photos were found", or something like that, appeared

**Reply by lastzero** [reply]
Maybe a feedback loop!? Logs get sent through websockets, so when those have issues and generate logs themselves it could lead to a loop in theory. Never saw that happening in practice though.

As for transcoding, it may be a problem with hardware transcoding then. Our preview should automatically use the software encoder as a fallback.

**Reply by Pheggas** [reply]
Right. What would be your strategy to fix this issue at this point? Maybe block the feedback look somehow?

Edit: [I described actual behavior on reddit](https://www.reddit.com/r/photoprism/comments/uotaed/megathread_photoprism_is_incredibly_slow_after/?utm_medium=android_app&utm_source=share)

**Reply by lastzero** [reply]
Wait, even without trace or debug mode it isn't working? Are you sure these are not background transcoding processes? Not working today, so can't take a deeper look.

**Reply by lastzero** [reply]
Did you restart photoprism or just the host VM?

**Reply by Pheggas** [reply]
> Wait, even without trace or debug mode it isn't working? Are you sure these are not background transcoding processes? Not working today, so can't take a deeper look.

Yep. I turned off both trace, debug and also checked if it isn't checked in Photoprism Web UI settings. Everything looked turned off, set to false so i restarted the Photoprism stack in Portainer and nothing really changed. 

I'm assuming this is really major issue and should be investigated as this can happen pretty much to anyone who use Photoprism.

**Reply by Pheggas** [reply]
> Did you restart photoprism or just the host VM?

I tried both actually. I tried restarting only Photoprism stack using Portainer and when I saw it didn't work I tried to restart whole VM. That didn't work neither.

**Reply by Pheggas** [reply]
I will keep whole Photoprism stack untouched, will not wipe it so we can maybe find the issue.

**Reply by lastzero** [reply]
Sorry, but you have a highly complex technical stack with Proxmox, VMs, Portainer.... and I don't even know yet which specific version of PhotoPrism you are using and if you have already tried to upgrade to Preview. You probably also have a reverse proxy like NGINX?

We can't provide support at this level in our community forum as we simply don't have the resources. You seem to expect us to ask for all the details from you first, or to keep guessing. The original question was about file format support. Then it turned out that you are using hardware transcoding in a multi-virtualized environment.

Just a few examples: How did you turn off trace mode? You have to remove the variable completely, while Portainer tends to remember variables once used. When you restart the container, you have to use stop and up, not stop and start. I'm not sure what Portainer's default actions are.

**Reply by Pheggas** [reply]
Right. I'm using Build 220302-0059f429-Linux-AMD64. I did not upgraded to preview yet as i was scared of corrupting whole database even more. I don't have set up any reverse proxy yet.

I understand your position. My best guess is this was caused by portainer itself. I don't have docker-compose installed on my VM and everything is managed through Portainer. Firstly, i turned on debug mode in Web UI. Then i added trace line in compose file in Portainer and stopped / started the stack. From now on, whole Photoprism was slow. I rewrote the "true" to "false" on trace line and again stopped / started the stack. Any move i did wasn't helpful. In fact, it seems "Albums" page in Web UI is loading instantly. But it is the only page that works normally.

My next moves: I will try to remove whole trace line entirely, stop / start the stack. When it wouldn't help i will try to stop / start each container. After then i will probably upgrade to preview and see if that helps. Even after then i will wipe whole thing.

**Reply by Pheggas** [reply]
Update: So i did the moves i described in comment above. Stopping and starting individual containers didn't help but what helped was changing the image from latest to preview.

I'm expecting some stability issues but i'm going to make backup ASAP so even if there will be some issues, i can always switch back to latest and recover the backup there. But before that, i have question about backup command.

In docks, there is `docker-compose exec photoprism photoprism backup -a --albums-path PATH -i --index-path PATH [FILENAME]` so i see the example executing command inside docker. But i'm not sure what is the first "photoprism". Is that name of the container in which the command will be executed? And then the second "photoprism" and right next "backup" is the actual comand inside the container?

So theoretically, if i would do something like `docker exec photoprism sh`, and when i will be inside the container, i would execute `photoprism backup ....` i will achieve the same result?
